### PR TITLE
Replace Debug.assert by standard JavaScript console.assert

### LINKED ISSUE #3088: Investigate how to strip Debug.assert calls from JavaScript
It would be good to figure how to remove the asserts from "production" builds. Different es6 module bundler provide different options. At the very least document how to do this with the most common ones.

### PR DESCRIPTION
**Pull request by pepone**
Fix #3088

### CODE REVIEW

**Review by externl — APPROVED** [review]
Looks good. I wonder however what other projects do in general. I've never used a dependency where I've ran into documentation about removing debug asserts. 

What's the best practice here in general.

For example our IceRPC docs website doesn't have anything for its webpack config. I haven't seen it mentioned looking at Webpack "best practices" pages. Are we maybe just not supposed to be doing this in general?

File: js/README.md [hunk 1]
```diff
@@ -44,9 +44,24 @@ finally
 }
 ```
 
+## Development
+
+The JavaScript code uses `console.assert` statements for debugging, labeled `DEV:` you should configure your production
+builds to remove such statements. Different module bundlers provide different options for example:
```
Comment (bernardnormier) [inline js/README.md:50]: ... labeled `DEV:`. You should configure ...

different options, for example:

File: js/README.md [hunk 2]
```diff
@@ -44,9 +44,24 @@ finally
...
+
+- If using [esbuild] see [drop-labels option][esbuild-drop-labels].
+- If using rollup see the [@rollup/plugin-strip] plugin and the [drop-labels option][strip-drop-labels].
+- If using WebPack the [TerserWebpackPlugin] plugin can be configured to drop `console.assert` see `drop_console` in [terser options].
```
Comment (bernardnormier) [inline js/README.md:54]: drop `...`; see ...

File: js/README.md [hunk 3]
```diff
@@ -44,9 +44,24 @@ finally
 }
 ```
 
+## Development
+
+The JavaScript code uses `console.assert` statements for debugging, labeled `DEV:` you should configure your production
+builds to remove such statements. Different module bundlers provide different options for example:
+
+- If using [esbuild] see [drop-labels option][esbuild-drop-labels].
+- If using rollup see the [@rollup/plugin-strip] plugin and the [drop-labels option][strip-drop-labels].
```
Comment (bernardnormier) [inline js/README.md:53]: if using xxx, see yyy

**Review by bernardnormier — APPROVED** [review]

**Comment by pepone** [comment]
> What's the best practice here in general.

I think the bundlers provide this options because it is something common enough. I personally prefer keep using asserts in our code, as it helps identify bugs.

**Pull request merged**